### ISSUE TITLE
Integrate KDK FAN

### ISSUE THREAD

**Issue by jovanyeo** [smartHomeHub/SmartIR#1054]
Dear SmartIR Expert,

First of all, thanks for great job.

However, I got an issue integration smart IR(FAN) with boardlink.

My issue is I have to press twice on the HA icon then my fan will switch on.

Sorry for my general description as I am not sure what will be the related information which you needed to assist me.

Looking forward to hear from you soon.

**Comment by jovanyeo** [comment]
{
  "manufacturer": "KDK",
  "supportedModels": [
    "M11SU"
  ],
  "supportedController": "Broadlink",
  "commandsEncoding": "Base64",
  "speed": [
    "low",
    "medium",
    "high"
  ],
  "commands": {
    "off": "JgDsAHA7DRANLA4PDRANDw8PDBEMEA4PDhALEQwRDQ8NLA4QDQ8RDAwWCBAODw4rDQ8OLA4rDw4PDwwtDg4RDA4PDhANKw0RCxEOEA4ODC0OLA0QDQ8OLA0RCxEPDwwtDCwODw4PDiwODw0sDg8PKg4QDSwODw0ACm9yOw0QDSwODw0QDhAMEQsUCREODw4PCxAOEA0PDiwMEQ4ODhAPDgwQDQ8OLA4PDS0NLQsRDxQJKwsRDhAMEA0PDi0QDA0QDBAPEAwsDSwODw4QDSwQDQ0QDBEOKw0sDg8ODw0sDhEMLA0QDS0NDw4rDg8OAA0F",
    "default": {
	  "on": "JgDsAHI5Dg8OLBEMEQsODw4PDg8NDw4PDg8RDBILDQ8OLBEMDg8NDw4PDg8ODw4rEgsTJw4rDg8ODw4sDRAQDA4PDg8RKQ0PDg8ODw4PDisOLBEMDg8NLBEMDg8ODw0sDiwODxEMEicODw4sDQ8OLBEMESgODw4ACnF2NxILDSwRDBEMEQwODw0PDg8ODw4PDg8NDxMKDiwODw0PDg8ODw4PDg8RKA4PDiwNLA4PDg8OKw4PEgsTCg4PDisSCw4PDg8ODw0sDiwRDBILDSwODw4PDg8NLBEpDg8NDw4sEQwOKw4PDiwODxEoDg8OAA0F",
      "low": "JgDsAHI6DQ8OLBEMDQ8ODw4PDg8NDw4PEgsTCg4PDQ8SKA4PDg8NDw4PDg8ODxEoDg8OLBInDg8ODw4sEQsTChEMDg8SJw4PDg8ODw4PDisSKA4PEQwQKREMDg8ODw0PDiwODw4PDSwODw4sDg8NDw4PDiwQDQ0ACqp3NhEMDSwRDBMKDRANDw4PDg8ODw0QEQsTCg4PDisSCw4PDg8ODw4PEQsOLBEMESgOLA4PDg8NLBMKDg8ODw0PEigODw4PDQ8ODw4sESgODw4PESkNDw4PDg8ODw4rEgsSCw4sDRANLA4PDg8RDA0sDg8RAA0F",
	  "medium": "JgDsAHA3EQwRKREMEAwRDBEMEA0QDRAMEQwQDRANDw0RKREMEA8ODBEMEQwQDRApEQwQKhApEQwQDRApEQwQDRANEA0QKRANEA0QDBEMESkQKREMEQwQDRApEQwRDBANECkRDBEMEQwQKREpEA0QDRANECkQDRAACqd1OBANECoPDRANEA0QDQ8ODw4PDRANEA0PDg8ODyoQDQ8ODw4PDRANEA0PKw8NECoPKw8NEA0QKg8ODw0QDRANECoPDRANEA0QDQ8qESkQDQ8ODw0RKRQJEA0QDBEpEA0QDRANECkQKhANEAwRDBAqEA0QAA0F",
      "high": "JgDsAHM3EQwRKRANEA0QDBEMEA0QDRANEAwRDBANEA0QKREMEA0QDRANEAwRDBAqEA0QKRAqEAwUCREpEA0QDBEMEA0QKhAMEQwQDRANDyoRKRANDw4PKhAqEA0QDBANECoQDQ8NECoPKhEpEA0PDg8ODyoQDQ8ACm91NxEMECoPDg8NEA0QDRANDw4PDRANEA0QDQ8ODyoQDRANDw4PDRANEA0QKhAMECoUJhMJEA0QKg8ODw0QDRANECoQDBANEA0QDQ8qESkQDQ8ODyoRKRANDw0QDRAqEA0PDREpECoQKRANEA0PDg8qEQwQAA0F"
      }
	 }
}

**Comment by jovanyeo** [comment]
@vassilis-panos care to have a look on my concern.

Many thanks in advance.

**Comment by jovanyeo** [comment]
@auanasgheps will you mind to assist me, please

**Comment by auanasgheps** [comment]
> My issue is I have to press twice on the HA icon then my fan will switch on.

Have you captured the command by yourself or are you using a template?

**Comment by jovanyeo** [comment]
> > My issue is I have to press twice on the HA icon then my fan will switch on.
> 
> Have you captured the command by yourself or are you using a template?

Captured the command by myself, thereafter, I added and edit them with one of your existing template.

**Comment by auanasgheps** [comment]
Try capturing the command again using the appropriate service.
Go to Developer Tools > Services.
Select Re_mote: Learn Command_.
Once the command is captured, use the service _Remote Service_ to send the learned command. Does it still require two presses?

**Comment by jovanyeo** [comment]
> Try capturing the command again using the appropriate service. Go to Developer Tools > Services. Select _Remote: Learn Command_. Once the command is captured, use the service _Remote Service_ to send the learned command. Does it still require two presses?

Let me try again.
By the way, I have no issue with the command if I don't use SmartIR.
The reason I use smartIR is because of the animination.

**Comment by jovanyeo** [comment]
> Try capturing the command again using the appropriate service. Go to Developer Tools > Services. Select _Remote: Learn Command_. Once the command is captured, use the service _Remote Service_ to send the learned command. Does it still require two presses?

Hihi,
Just wanna to ask, are trying to ask to perform the following,:
1) Re-learn the code again
2) Put the new re-learned code to the smartIR,Jason file?
3) and test it again.
